msteams-platform/messaging-extensions/build-api-troubleshoot.md

```diff
@@ -0,0 +1,93 @@
+---
+title: API-based message extension FAQ
+author: v-ypalikila
+description: Learn how to troubleshoot common issues encountered with API Message Extensions. 
+ms.localizationpriority: medium
+ms.topic: overview
+ms.author: anclear
+ms.date: 03/27/2024
+---
+
+# API-based message extensions FAQ
+
+Get answers to common questions about API-based message extensions.
+<br>
+
+<details><summary> Q. What are the key elements to check in an app manifest for a successful deployment?</summary>
+
+* Make sure that the app manifest version is "devPreview"
+* The paths to the apiSpecificationFile and apiResponseRendering template must be relative paths.
+* Make sure that the command Ids match exactly with the operation Ids defined in the OpenAPI spec file
+* Make sure that the parameters for each command match exactly with the names of the parameters defined for the operation in the OpenAPI spec.
+
+</details>
+<br>
+
+<details><summary> Q. What are the best practices for troubleshooting issues with an OpenAPI spec?</summary>
+
+Make sure that the OpenAPI spec adheres to the limitations defined in [build API-based message extension](build-api-based-message-extension.md#OAD).
+
+</details>
+<br>
+
+<details><summary> Q. How do you ensure that a response rendering template is correctly formatted and functional?</summary>
+
+Call the API using Fiddler or Postman to make sure the requests look correct, and the response is valid. Also grab a sample response to use for validating the response rendering template. Teams only supports up to version 1.5 for adaptive cards but the designer supports up to version 1.6.
```

Comment:
@maoyu-dev we can't leave it like this. This is the equivalent of

> Q: How do I ensure my vacuum cleaner is working?
> A: Open your vacuum cleaner and ensure that everything is working correctly.

I mean, sure it's *not wrong*, but it's not helpful either. We need to say more about *how* to do this.

Reply:
The visual studio code teams toolkit actually has a visualizer for this. We can tell them to use that. Though that really only works for the vscode path

Reply:
Removed the FAQ page and added a new section for [Troubleshooting](https://review.learn.microsoft.com/en-us/microsoftteams/platform/messaging-extensions/build-api-based-message-extension?branch=pr-en-us-10529&tabs=token-v2#troubleshooting).